packages/clay-button/src/ClayButton.soy

```diff
@@ -57,20 +57,42 @@
 	{@param? label: html|string}
 	{@param? spritemap: string}
 
+	{let $classes kind="text"}
+		inline-item
+		{if $iconAlignment == 'left' and $label}
+			{sp}inline-item-before
+		{elseif $iconAlignment == 'right' and $label}
+			{sp}inline-item-after
+		{/if}
+	{/let}
+
+	{let $iconContent kind="html"}
```

Comment:
Refactor all this, please. You're defining `$iconContent` and `$iconWrapper` even when they would be not used.